### DISCUSSION TITLE
Comprehension lists

### DISCUSSION THREAD

**Discussion by manuel-rubio** [gleam-lang/gleam#2175] [Ideas & suggestions]
As a good Erlanger, I like the possibility to perform this kind of action. I know that it's like a shortcut for something we could do with `list.map/2` and `list.filter/2`, but I think that's very expressive and convenient.

It could be implemented in the Erlang way:
```
[ car || car <- cars, car.country == "DE" ]
```
or it could be implemented in the Elixir (and almost Python) way:
```
for car <- cars if car.country == "DE" { car }
```
what do you think?

**Comment by lpil** [comment]
In Gleam we avoid having multiple ways to do things, and comprehensions don't enable the programmer to do anything new, so I don't think they would be a high value addition.

Having said that, you can achieve the same effect with use expressions.

```gleam
let pairs = {
  use x <- list.flat_map(xs)
  use y <- list.map(ys)
  #(x, y)
}
```
